### PR TITLE
archi: fix gtk errors

### PR DESCRIPTION
**Pull request by paumr**
###### Description of changes
This PR fixes some issues (missing dependencies) which caused Archi to crash.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by earldouglas — APPROVED** [review]

File: pkgs/tools/misc/archi/default.nix [hunk 1]
```diff
@@ -43,6 +46,7 @@ stdenv.mkDerivation rec {
 
         install -D -m755 Archi $out/libexec/Archi
         makeWrapper $out/libexec/Archi $out/bin/Archi \
+          --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath ([ webkitgtk ])} \
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/archi/default.nix:49]: Can we rather patchelf?

Comment (paumr) [reply pkgs/tools/misc/archi/default.nix:49]: Are you sure that patchelf is viable here?
This package is quite complex, i fear we'd have to change a lot.

Eclipse seems to solve it similairly (this package runs on the same platform):
https://github.com/NixOS/nixpkgs/blob/6c9e17080423bcba9f09ebbcd5ffc55e34af2449/pkgs/applications/editors/eclipse/build-eclipse.nix#L45

Comment (SuperSandro2000) [reply pkgs/tools/misc/archi/default.nix:49]: Can you give it a try? Changing LD_LIBRARY_PATH affects alls applications started within the program while patchelf is only for the program itself.

Comment (paumr) [reply pkgs/tools/misc/archi/default.nix:49]: @SuperSandro2000 
i did, thus the question.
`patchelf` won't be enough, we'd have to patch some of the jar files.

Thinking of it - passing the library to it's childen might actually be the desired behavior - unless we patch third-party plugins too.

**Pull request merged**